app/src/main/java/it/ministerodellasalute/immuni/ui/upload/UploadDataFragment.kt

```diff
@@ -66,7 +68,9 @@ class UploadDataFragment : Fragment(R.layout.upload_data_fragment), Confirmation
         })
 
         upload.setSafeOnClickListener {
-            uploadOtp()
+            GlobalScope.launch {
```

Comment:
Is there a specific reason why you introduced this `GlobalScope` non lifecycle aware usage?